src/server/failover.c

```diff
@@ -806,6 +810,142 @@ alt_conn(pbs_net_t addr, unsigned int sec)
 
 /**
  * @brief
+ * 	Function to check if stonith script exists at PBS_HOME/server_priv/stonith.
+ * 	If it does then invoke the script for execution.
+ * 
+ * @param[in]	node - hostname of the node, that needs to be powered off.
+ * 
+ * @return	Error code
+ * @retval	 0 - stonith script executed successfully or script does not exist.
+ * @retval      -1 - stonith script failed to bring down node.
+ *
+ */
+int
+check_and_invoke_stonith(char *node)
+{
+	char		stonith_cmd[3*MAXPATHLEN+1] = {0};
+	char		stonith_fl[MAXPATHLEN+1] = {0};
+	char		*p = NULL;
+	char		out_err_fl[MAXPATHLEN+1] = {0};
+	char		*out_err_msg = NULL;
+	int		rc = 0;
+	int		fd = 0;
+	struct stat	stbuf;
+#ifdef WIN32
+	char buf[MAXPATHLEN+1] = {0};
```

Comment:
You can go for a more meaningful name if `buf` is used to hold just the stonith script location.

Reply:
Have changed it.